include/libtorrent/torrent_handle.hpp

```diff
@@ -680,23 +680,51 @@ namespace aux {
 		// around separately, or for torrents that were added via a magnet link.
 		static constexpr resume_data_flags_t save_info_dict = 1_bit;
 
-		// if nothing significant has changed in the torrent since the last
-		// time resume data was saved, fail this attempt. Significant changes
-		// primarily include more data having been downloaded, file or piece
-		// priorities, paused-state, transfer limit settings  having changed etc.
-		// Counters keeping statistics are notably *not* considered significant.
-		// Things like total_uploaded, total_downloaded, last_seen_complete,
-		// tracker scrape counters, are not considered significant enough
-		// changes for this flag to save the resume data.
-		// If the resume data doesn't need saving, a
-		// save_resume_data_failed_alert is posted with the error
-		// resume_data_not_modified.
+		// this flag has the same behavior as the combination of:
+		// if_counters_changed | if_download_progress | if_config_changed |
+		// if_state_changed | if_metadata_changed
 		static constexpr resume_data_flags_t only_if_modified = 2_bit;
 
-		// this flag amends the behavior of only_if_modified to include
-		// stats counter or scrape data updating to also count as an update, so
-		// it becomes more eager to save resume data.
-		static constexpr resume_data_flags_t save_counters = 3_bit;
+		// save resume data if any counters has changed since the last time
+		// resume data was saved. This includes upload/download counters, active
+		// time counters and scrape data. A torrent that is not paused will have
+		// its active time counters incremented continuously.
+		static constexpr resume_data_flags_t if_counters_changed = 3_bit;
+
+		// save the resume data if any blocks have been downloaded since the
+		// last time resume data was saved. This includes:
+		// * checking existing files on disk
+		// * downloading a block from a peer
+		// * files renamed or moved
```

Comment:
Does this refer to the download progress and not to the state?

Reply:
yeah, I'm unsure about this. "moved" should probably be considered the "save_path" state. but it's possible that when moving the storage new files will be picked up. However, I suppose that case would be covered already, by checking files.

Reply:
>I suppose that case would be covered already, by checking files.

👍